fe/fe-core/src/main/java/com/starrocks/leader/ReportHandler.java

```diff
@@ -456,6 +479,33 @@ private static void workgroupReport(long backendId, List<TWorkGroup> workGroups)
                 backendId, System.currentTimeMillis() - start, workGroups.size());
     }
 
+    // For test.
+    public static void testHandleResourceUsageReport(long backendId, TResourceUsage usage) {
+        resourceUsageReport(backendId, usage);
+    }
+
+    private static void resourceUsageReport(long backendId, TResourceUsage usage) {
+        LOG.debug("begin to handle resource usage report from backend {}", backendId);
+        long start = System.currentTimeMillis();
+        Backend backend = GlobalStateMgr.getCurrentSystemInfo().getBackend(backendId);
+        if (backend == null) {
+            LOG.warn("backend doesn't exist. id: {}", backendId);
+            return;
+        }
+
+        boolean isChanged = backend.updateResourceUsage(usage.getNum_running_queries(), usage.getMem_limit_bytes(),
+                usage.getMem_used_bytes());
+        if (isChanged) {
+            LOG.debug("resource usage from backend {} has changed, and add it to edit log", backendId);
+            GlobalStateMgr.getCurrentState().getEditLog().logUpdateResourceUsage(
```

Comment:
Will it increase pressure of editlog?

Reply:
When there are lots of BEs, maybe it does. Therefore, sync resource usage to FE followers by RPC instead of EditLog.